lib/chat_web/live/main_live/layout/secret_sharing_form.ex

```diff
@@ -0,0 +1,215 @@
+defmodule ChatWeb.MainLive.Layout.SecretSharingForm do
+  @moduledoc """
+  Handles showing and updating secret sharing as a setting
+  """
+
+  use ChatWeb, :live_component
+
+  import Phoenix.Component
+
+  alias Chat.Card
+  alias Chat.Dialogs.{Dialog, DialogMessaging}
+  alias Chat.Messages.Text
+  alias Chat.User.Registry
+
+  @impl Phoenix.LiveComponent
+  def mount(socket) do
+    users =
+      Registry.all()
+      |> Stream.map(fn {pub_key, %Card{} = user} -> {pub_key, user.name} end)
+      |> Enum.to_list()
+      |> Enum.sort_by(&elem(&1, 1))
```

Comment:
1. This list includes myself, which is wrong
2. `UsersBrocker` is a better source of users